trial-config-service/src/main/java/uk/ac/ox/ndph/mts/trial_config_service/controller/TrialConfigController.java

```diff
@@ -12,95 +14,89 @@
 import org.springframework.web.bind.annotation.RestController;
 import org.springframework.web.reactive.function.client.WebClient;
 import org.springframework.web.reactive.function.client.WebClientException;
-import reactor.core.publisher.Mono;
-import uk.ac.ox.ndph.mts.trial_config_service.config.GitRepo;
 import uk.ac.ox.ndph.mts.trial_config_service.exception.InvalidConfigException;
 import uk.ac.ox.ndph.mts.trial_config_service.exception.ResourceAlreadyExistsException;
 import uk.ac.ox.ndph.mts.trial_config_service.model.Trial;
 import uk.ac.ox.ndph.mts.trial_config_service.service.TrialConfigService;
+import java.io.File;
 
-import java.io.IOException;
-
-
-
+@RefreshScope
 @RestController
 @RequestMapping("/trial-config")
 public class TrialConfigController {
 
+    @Value("${spring.cloud.config.uri}")
+    private String configURI;
+
+    @Value("${spring.application.name}")
+    private String applicationName;
+
+    @Value("${spring.cloud.config.profile}")
+    private String profile;
+
+    @Value("${spring.cloud.config.label}")
+    private String label;
+
     private final TrialConfigService trialConfigService;
     private final WebClient webClient;
-    private GitRepo gitRepo;
 
     //TODO(darrensmithson) - only temp until we decide where to get userId from
     String userId = "admin001";
 
     @Autowired
-    public TrialConfigController(TrialConfigService trialConfigService, WebClient webClient, GitRepo gitRepo) {
+    public TrialConfigController(TrialConfigService trialConfigService, WebClient webClient) {
         this.trialConfigService = trialConfigService;
         this.webClient = webClient;
-        this.gitRepo = gitRepo;
-    }
-
-    public TrialConfigController(TrialConfigService trialConfigService, String baseUrl) {
-        this.trialConfigService = trialConfigService;
-        this.webClient = WebClient.create(baseUrl);
     }
 
     //TODO(darrensmithson) - we are using a GET and POST for similar functionlaity to accomodate
     // early testing needs, once we are closer to final developmemnt we should ensure only 1 POST is used.
     @GetMapping("/trial")
-    public Trial saveTrialFromGit(
+    public Trial saveTrialFromConfigServer(
```

Comment:
I prefer all changes to trial config server will be reverted.
a) its going refactoring these days. so we dont want to conflict
b) this method should be deleted. the trial conf service is not exposing endpoint (this will be part of the refactor)

Reply:
I think if we decide to remove any existing endpoints it needs to be a separate unit of work - we should keep this endpoint as it is for this story. The renaming of the method is clarification that we are now not u sing the gitrepo directly but using the config server - changing the name is purely for readability purposes as it will never be called - only the endpoint..

Reply:
Also just to clarify - the refactoring is necessary to get the controllers working with the config server - when we merge to main these changes will be needed.

Reply:
yes I agree. my point is that the team is already working on the init service (trial config service). so all these changes should be reverted.

Reply:
I think they should go into 341 and then we make a decision to include them in main - keeping the changes in the controllers lets the devs know how to make the config server 'work'. If we decide against merging them into main we should keep the branch for reference, that said, if we are to rework trial-config server why not keep the changes any way as they will be refactored - or possibly completely deleted? I think we need a ref how we retrieve the files..it will save some time for somebody.